src/blurredLocation.js

```diff
@@ -97,11 +98,28 @@ BlurredLocation = function BlurredLocation(options) {
     options.map.panTo(new L.LatLng(lat, lng));
   }
 
-  function getPlacenameFromCoordinates(lat, lng, onResponse) {
+  function getPlacenameFromCoordinates(lat, lng, precision, onResponse) {
       $.ajax({
-      url:"https://maps.googleapis.com/maps/api/geocode/json?latlng="+lat+","+lng,
-      success: function(result) {
-        onResponse(result);
+        url:"https://maps.googleapis.com/maps/api/geocode/json?latlng="+lat+","+lng,
+        success: function(result) {
+          if(result.results[0]) {
+            var country;
+            var fullAddress = result.results[0].formatted_address.split(",");
+            for (i in result.results) {
+              if(result.results[i].types.indexOf("country") != -1) {
+                //If the type of location is a country assign it to thr input box value
+                country = result.results[i].formatted_address;
+              }
+            }
+
+            if(precision <= 0) onResponse(country);
+
+            else if(precision == 1) onResponse(fullAddress[fullAddress.length - 2] + "," + country);
+
+            else if(precision >= 2) onResponse(fullAddress[fullAddress.length - 3] + "," + fullAddress[fullAddress.length - 2] + "," + country);
```

Comment:
We should also ensure that if precision is not set, there's a default -- probably just the full address.